tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,266 @@
+                        .AcquireTokenForClient(s_msi_scopes)
+                        .WithManagedIdentity(userIdentity)
+                        .ExecuteAsync().ConfigureAwait(false);
+            
+                //Assert
+                //1. Access token
+                Assert.IsNotNull(result.AccessToken);
+
+                //2. First token response is from the MSI Endpoint
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                //3. Validate the ExpiresOn falls within a 24 hour range from now
+                CoreAssert.IsWithinRange(
+                                DateTimeOffset.UtcNow + TimeSpan.FromHours(0),
+                                result.ExpiresOn,
+                                TimeSpan.FromHours(24));
+
+                result = await cca
+                    .AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                //4. Validate the scope
+                Assert.IsTrue(s_msi_scopes.All(result.Scopes.Contains));
+
+                //5. Validate the second call to token endpoint gets returned from the cache
+                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow(MsiAzureResource.WebApp, NonExistentUserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(MsiAzureResource.WebApp, Non_Existent_Mi_res_id, DisplayName = "ResourceID Web App")]
+        public async Task MSIWrongClientIDAsync(MsiAzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            using (new EnvVariableContext())
+            {
+                //Get the Environment Variables
+                Dictionary<string, string> envVariables =
+                    await GetEnvironmentVariablesAsync(azureResource).ConfigureAwait(false);
+
+                //Set the Environment Variables
+                SetEnvironmentVariables(envVariables);
+
+                //form the http proxy URI 
+                string uri = s_baseURL + $"MSIToken?" +
+                    $"azureresource={azureResource}&uri=";
+
+                //Create CCA with Proxy
+                IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+                //Act
+                MsalServiceException ex = await AssertException.TaskThrowsAsync<MsalServiceException>(async () =>
+                {
+                    await cca
+                    .AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync().ConfigureAwait(false);
+                }).ConfigureAwait(false);
+
+                //Assert
+                Assert.IsTrue(ex.Message.Contains(UserAssignedIdDoesNotExist));
```

Comment:
Assert error code instead of error message? Since the error message is based on the server response.

Reply:
this seems to be a very generic error code but the message seems to be specific in this scenario. thoughts? 

![image](https://user-images.githubusercontent.com/90415114/224449415-f672b172-2164-4435-bdc8-317aaa4e1a72.png)

Reply:
I am thinking how will asserting this message help us? Are we going to take some action if the error message changes?

Reply:
`Assert.InformButDontFail` :)

Reply:
this message is from the Web App Azure Resource when you pass a wrong client id, this is not from the helper service. Helper service does not manipulate any error codes or error messages it simply passes it down

![image](https://user-images.githubusercontent.com/90415114/225173034-9e5e4eba-812b-4f61-8c3f-f5eccfce8cfe.png)